### PR TITLE
44229 manage user

### PR DESCRIPTION
**Pull request by abdullahallmehedi**

### CODE REVIEW

File: app/views/users/index.html.erb [hunk 1]
```diff
@@ -1,9 +1,13 @@
...
+<h1><%= t :users_title %></h1>
 
 <%= will_paginate %>
 
-<ul class="users">
-  <%= render @users %>
+<ul class="users">  
+  <% @users.each do |user| %>
+  <li>
+  	<a href="<%= user_path user.id %>"><%= user.lastname %>, <%= user.firstname %></a>
```
Comment (chuanhtuan) [inline app/views/users/index.html.erb]: link_to

File: app/views/users/show.html.erb [hunk 2]
```diff
@@ -5,12 +5,9 @@
...
       </h1>
     </section>
-    <section class="stats">
-      <%= render 'shared/stats' %>
-    </section>
   </aside>
   <div class="col-md-8">    
-    <% if @user.activities.any? %>
+    <% if @activities.any? %>
       <h3>Activities (<%= @user.activities.count %>)</h3>
```
Comment (chuanhtuan) [inline app/views/users/show.html.erb]: i18n

File: app/views/admin/users/show.html.erb [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+  <aside class="col-md-4">
+    <section class="user_info">
+      <h1>        
+        <%= @user.firstname %> <%= @user.lastname %>
+      </h1>
+    </section>
+  </aside>
+  <div class="col-md-8">
+    <% if @activities.any? %>
+      <h3>Activities (<%= @user.activities.count %>)</h3>
```
Comment (chuanhtuan) [inline app/views/admin/users/show.html.erb]: i18n

File: app/views/admin/users/index.html.erb [hunk 4]
```diff
@@ -0,0 +1,14 @@
+<h1><%= t :users_title %></h1>
+
+<%= will_paginate %>
+
+<ul class="users">  
+  <% @users.each do |user| %>
+  <li>
+  	<a href="<%= admin_user_path user.id %>"><%= user.lastname %>, <%= user.firstname %></a>
```
Comment (chuanhtuan) [inline app/views/admin/users/index.html.erb]: link_to

File: app/views/admin/users/index.html.erb [hunk 5]
```diff
@@ -0,0 +1,14 @@
+<h1><%= t :users_title %></h1>
+
+<%= will_paginate %>
+
+<ul class="users">  
+  <% @users.each do |user| %>
+  <li>
+  	<a href="<%= admin_user_path user.id %>"><%= user.lastname %>, <%= user.firstname %></a> 
+  	<a href="<%= edit_admin_user_path user.id %>"><%= t(:update) %></a>
```
Comment (chuanhtuan) [inline app/views/admin/users/index.html.erb]: link_to

**Comment by chikim** [comment]
2 commits, pls rebase

File: app/controllers/admin/users_controller.rb [hunk 6]
```diff
@@ -0,0 +1,30 @@
+class Admin::UsersController < ApplicationController
+  before_action :logged_in_user
+  def index
```
Comment (chikim) [inline app/controllers/admin/users_controller.rb]: there should be an empty line to seperate callbacks and functions declaration

File: app/controllers/admin/users_controller.rb [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+    @users = User.paginate page: params[:page]
+  end
+  
+  def show
+    @user = params.has_key?(:id) ? User.find(params[:id]) : current_user
```
Comment (chikim) [inline app/controllers/admin/users_controller.rb]: use before_action

File: app/controllers/admin/users_controller.rb [hunk 8]
```diff
@@ -0,0 +1,30 @@
...
+    @users = User.paginate page: params[:page]
+  end
+  
+  def show
+    @user = params.has_key?(:id) ? User.find(params[:id]) : current_user 
+    @activities = @user.activities.paginate page: params[:page]
+  end
+  
+  def edit
+    @user = User.find params[:id]
```
Comment (chikim) [inline app/controllers/admin/users_controller.rb]: use before_action

File: app/controllers/admin/users_controller.rb [hunk 9]
```diff
@@ -0,0 +1,30 @@
...
+  end
+  
+  def update
+    @user = User.find params[:id]
```
Comment (chikim) [inline app/controllers/admin/users_controller.rb]: use before_action

File: app/helpers/users_helper.rb [hunk 10]
```diff
@@ -0,0 +1,2 @@
+module UsersHelper
```
Comment (chikim) [inline app/helpers/users_helper.rb]: remove unused file

File: app/views/admin/users/index.html.erb (hunk 4 above)
Comment (chikim) [inline app/views/admin/users/index.html.erb]: indent

File: app/views/admin/users/index.html.erb (hunk 5 above)
Comment (chikim) [inline app/views/admin/users/index.html.erb]: indent

File: app/views/admin/users/index.html.erb [hunk 11]
```diff
@@ -0,0 +1,14 @@
+<h1><%= t :users_title %></h1>
+
+<%= will_paginate %>
+
+<ul class="users">  
+  <% @users.each do |user| %>
+  <li>
+  	<a href="<%= admin_user_path user.id %>"><%= user.lastname %>, <%= user.firstname %></a> 
+  	<a href="<%= edit_admin_user_path user.id %>"><%= t(:update) %></a> 
+  	</li>
```
Comment (chikim) [inline app/views/admin/users/index.html.erb]: indent

File: app/views/users/index.html.erb (hunk 1 above)
Comment (chikim) [inline app/views/users/index.html.erb]: indent

**Comment by chikim** [comment]
LGTM

File: config/routes.rb [hunk 12]
```diff
@@ -8,8 +8,10 @@
   resources :authentications, only: [:new, :create, :destroy]  
   resources :categories, only: :index
   namespace :admin do
+    root 'users#show'
```
Comment (chuanhtuan) [inline config/routes.rb]: you have to define resources user for admin namespace

**Pull request merged**